service/controller/clusterapi/v29/key/common.go

```diff
@@ -31,6 +31,10 @@ func IsDeleted(getter DeletionTimestampGetter) bool {
 	return getter.GetDeletionTimestamp() != nil
 }
 
+func MachineDeploymentASGName(getter LabelsGetter) string {
+	return fmt.Sprintf("asg-%s-tcnp", ClusterID(getter))
```

Comment:
```suggestion
	return fmt.Sprintf("asg-%s-tcnp-%s", ClusterID(getter), MachineDeploymentID(getter))
```
No strong feelings about the string in general, but I thought it makes sense to identify it by cluster and machine deployment (as there will be several machine deployments per cluster).

Reply:
Good point. I added some version of this.